pkg/tsdb/postgres/macros.go

```diff
@@ -79,11 +79,11 @@ func (m *PostgresMacroEngine) evaluateMacro(name string, args []string) (string,
 		}
 		return fmt.Sprintf("extract(epoch from %s) as \"time\"", args[0]), nil
 	case "__timeFilter":
-		// dont use to_timestamp in this macro for redshift compatibility #9566
 		if len(args) == 0 {
 			return "", fmt.Errorf("missing time column argument for macro %v", name)
 		}
-		return fmt.Sprintf("extract(epoch from %s) BETWEEN %d AND %d", args[0], m.TimeRange.GetFromAsSecondsEpoch(), m.TimeRange.GetToAsSecondsEpoch()), nil
+
+		return fmt.Sprintf("%s BETWEEN '%s' AND '%s'", args[0], m.TimeRange.MustGetFrom().UTC().Format(time.RFC3339), m.TimeRange.MustGetTo().UTC().Format(time.RFC3339)), nil
```

Comment:
A bit unsure of the `time.RFC3339`. Would that format be supported by all postgres versions? Or would `'<datetime formated>'::date` be better?

Also, would like `m.TimeRange.MustGetFrom().UTC()` to be encapsulated in `m.TimeRange.GetFromAsTimeUTC()` or something similar.

Reply:
I checked on 8.0.2 which is 13 years old and what redshift is based on and it worked there. So its safe to assume any recent postgres supports it. I choose this format cause its a standard and has the timezone in it aswell.